src/helpers/createTestHarness.tsx

```diff
@@ -1,42 +1,65 @@
 import React, { Suspense } from 'react'
+import { ErrorBoundary, FallbackProps } from 'react-error-boundary'
+import filterConsole from 'filter-console'
 
-import { RendererProps, WrapperComponent } from '../types/react'
+import { addCleanup } from '../core'
 
-import { isPromise } from './promises'
+import { RendererProps, WrapperComponent } from '../types/react'
 
-function TestComponent<TProps, TResult>({
-  hookProps,
-  callback,
-  setError,
-  setValue
-}: RendererProps<TProps, TResult> & { hookProps?: TProps }) {
-  try {
-    // coerce undefined into TProps, so it maintains the previous behaviour
-    setValue(callback(hookProps as TProps))
-  } catch (err: unknown) {
-    if (isPromise(err)) {
-      throw err
-    } else {
-      setError(err as Error)
+function suppressErrorOutput() {
+  // The error output from error boundaries is notoriously difficult to suppress.  To save
+  // out users from having to work it out, we crudely suppress the output matching the patterns
+  // below.  For more information, see these issues:
+  //   - https://github.com/testing-library/react-hooks-testing-library/issues/50
+  //   - https://github.com/facebook/react/issues/11098#issuecomment-412682721
+  //   - https://github.com/facebook/react/issues/15520
+  //   - https://github.com/facebook/react/issues/18841
```

Comment:
I wonder if we need all this information here or is it just noise to anyone else coming in after that change is made?

Reply:
No, keep it here, this some of the issues you posted are still open so I imagine *if* they do something to solve these, then it'll have an impact, i like the paper trail 🔍